src/Microsoft.ML.Tokenizers/Model/TiktokenTokenizer.cs

```diff
@@ -1209,7 +1216,30 @@ private static TiktokenTokenizer CreateForModel(
                     tiktokenConfiguration.VocabFile,
                     out (Dictionary<ReadOnlyMemory<byte>, int> encoder, Dictionary<StringSpanOrdinalKey, (int Id, string Token)> vocab, Dictionary<int, ReadOnlyMemory<byte>> decoder) cache))
             {
-                using Stream compressedStream = typeof(Tokenizer).Assembly.GetManifestResourceStream(tiktokenConfiguration.VocabFile)!;
+                Debug.Assert(tiktokenConfiguration.PackageName is not null);
+
+                string fullAssemblyName = typeof(TiktokenTokenizer).Assembly.FullName!;
+                int commaIndex = fullAssemblyName.IndexOf(',');
+
+                Assembly? assembly = null;
+                if (commaIndex > 0)
```

Comment:
I wonder if we should test in a linked app.  I can't  remember if this metadata is persisted or not.

Reply:
I did a quick try creating an app and publishing it using `<PublishAot>true</PublishAot>`. Then running code like `Console.WriteLine(typeof(Tokenizer).Assembly.FullName);` and that produced `Microsoft.ML.Tokenizers, Version=1.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51`

Reply:
Actuall the native compilation couldn't load the data assembly at runtime. I'll try to talk to the AoT guys to figure out if apps need to explicitly persist the missing data.  

```
Unhandled Exception: System.InvalidOperationException: The tokenizer data file Microsoft.ML.Tokenizers.Data.Cl100kBase.dll could not be loaded. Please reference the package Microsoft.ML.Tokenizers.Data.Cl100kBase in your project.
 ---> System.IO.FileNotFoundException: Cannot load assembly 'Microsoft.ML.Tokenizers.Data.Cl100kBase, Version=1.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. No metadata found for this assembly.
```

Reply:
Perf offline discussion we decided to go with using `Type.GetType` to avoid getting any trimming linker warning or any runtime failure. The commit https://github.com/dotnet/machinelearning/pull/7248/commits/6dfb2cfa3800617a4edb0eb22044ea0fe2e57031 implemented that.